Research/start.py

```diff
@@ -29,24 +29,66 @@
 
 from clr import AddReference
 AddReference("System")
-AddReference("QuantConnect.Algorithm")
-AddReference("QuantConnect.Api")
-AddReference("QuantConnect.Common")
-AddReference("QuantConnect.Configuration")
-AddReference("QuantConnect.Research")
-AddReference("QuantConnect.Indicators")
-
-from System import *
-from QuantConnect import *
-from QuantConnect.Algorithm import *
-from QuantConnect.Api import *
-from QuantConnect.Configuration import *
-from QuantConnect.Data import *
-from QuantConnect.Research import *
-from QuantConnect.Indicators import *
+
+#Load assemblies
+file_absolute_path = os.path.abspath(__file__)
+for file in os.listdir(os.path.dirname(file_absolute_path)):
+    if file.endswith(".dll"):
+        AddReference(file.replace(".dll", ""))
+
+from System  import *
+from System.Collections  import *
+from System.Collections.Generic  import *
+from System.Linq  import *
+from System.Globalization  import *
+from QuantConnect  import *
+from QuantConnect.Algorithm  import *
+from QuantConnect.Algorithm.Framework  import *
+from QuantConnect.Algorithm.Framework.Selection  import *
+from QuantConnect.Algorithm.Framework.Alphas  import *
+from QuantConnect.Algorithm.Framework.Portfolio  import *
+from QuantConnect.Algorithm.Framework.Execution  import *
+from QuantConnect.Algorithm.Framework.Risk  import *
+from QuantConnect.Api  import *
+from QuantConnect.Parameters  import *
+from QuantConnect.Benchmarks  import *
+from QuantConnect.Brokerages  import *
+from QuantConnect.Util  import *
+from QuantConnect.Interfaces  import *
+from QuantConnect.Indicators  import *
+from QuantConnect.Research  import *
+from QuantConnect.Data  import *
+from QuantConnect.Data.Consolidators  import *
+from QuantConnect.Data.Custom  import *
+from QuantConnect.Data.Fundamental  import *
+from QuantConnect.Data.Market  import *
+from QuantConnect.Data.UniverseSelection  import *
+from QuantConnect.Notifications  import *
+from QuantConnect.Orders  import *
+from QuantConnect.Orders.Fees  import *
+from QuantConnect.Orders.Fills  import *
+from QuantConnect.Orders.Slippage  import *
+from QuantConnect.Scheduling  import *
+from QuantConnect.Securities  import *
+from QuantConnect.Securities.Equity  import *
+from QuantConnect.Securities.Forex  import *
+from QuantConnect.Securities.Interfaces  import *
+from QuantConnect.Configuration  import *
 
 # Start an instance of an API class
 api = Api()
 api.Initialize(Config.GetInt("job-user-id", 1), 
     Config.Get("api-access-token", "default"),
     Config.Get("data-folder"))
+
+# Loads composer so that we do not get an exception in the jupyter notebook
+Composer.Instance
+
+#Site packages path needs to be added back for jupyter
+# otherwise Lean engine cannot import pandas, etc.
+import sys
+import site
+site_packages_path = site.getsitepackages()
+for site_package_path in site_packages_path:
+    if site_package_path not in sys.path:
+        sys.path.append(site_package_path)
```

Comment:
Calling `Composer.Instance` shouldn't be required, nor adding the site packages into the system path, believe this is due to some environment specific issue, this is why we push to use the new Lean CLI instead, which solves issues in a standard way.